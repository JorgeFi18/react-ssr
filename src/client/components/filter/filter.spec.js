import React from 'react';
import renderer from 'react-test-renderer';
import Filter from './filter.component';

describe('Filter component', () => {
    it('Should return -name value', () => {
        let value = '';
        const changeOption = e => {value = e};

        const component = renderer.create(
            <Filter filterBy={value} onFilterChange={changeOption}/>
        )

        let tree = component.toJSON();
        const [ , select ] = tree.children;
        select.props.onChange('-name');

        expect(value).toBe('-name');
    });

    it('Should check select default value', ()=> {

        const component = renderer.create(
            <Filter filterBy='modified' />
        )

        let tree = component.toJSON();
        const [ , select ] = tree.children;

        expect(select.props.value).toBe('modified');
    });

    it('Should match with snapshot', ()=> {
        const component = renderer.create(
            <Filter filterBy='name' />
        )

        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});