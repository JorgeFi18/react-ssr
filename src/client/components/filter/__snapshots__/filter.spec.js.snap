// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component Should match with snapshot 1`] = `
<div
  className="filter"
>
  <p
    className="filter__label"
  >
    Filter by:
  </p>
  <select
    className="filter__input"
    value="name"
  >
    <option
      value="name"
    >
      Name (ASC)
    </option>
    <option
      value="modified"
    >
      Date (ASC)
    </option>
    <option
      value="-name"
    >
      Name (DESC)
    </option>
    <option
      value="-modified"
    >
      Date (DESC)
    </option>
  </select>
</div>
`;
